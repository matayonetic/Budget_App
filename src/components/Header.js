import React, { useEffect } from "react";
import { startLogout } from "../actions/auth";
import { useDispatch } from "../react-redux-hooks";
import { NavLink, useNavigate } from "react-router-dom";
import { auth } from "../../src/firebase/firebase";

// Header
const Header = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();

  // Start Logout
  const beginLogout = () => {
    dispatch(startLogout());
  };

  // Check Login Status
  useEffect(() => {
    auth.onAuthStateChanged((user) => {
      if (!user) {
        console.log("User is currently logged out");
        navigate("/login");
      }
    });
  }, [auth]);

  // Check Log Status
  const logStatus = !!auth.currentUser;

  return (
    <header>
      <h1>Budget App</h1>
      <NavLink to={"/dashboard"}>Home</NavLink> &nbsp;
      <NavLink to={"/create"}>Add Expense</NavLink> &nbsp;
      <NavLink to={"/help"}>Help</NavLink> &nbsp;
      {logStatus ? <button onClick={beginLogout}>Logout</button> : undefined}
    </header>
  );
};

export default Header;
