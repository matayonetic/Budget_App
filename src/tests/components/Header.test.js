import React from "react";
import { shallow } from "enzyme";
import Header from "../../components/Header";

// Render Header
test("Should correctly render Header component", () => {
  const wrapper = shallow(<Header />);
  expect(wrapper).toMatchSnapshot();
});

/*
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Header from "../../components/Header";

// Render Header
test("Should render header correctly", () => {
  const renderer = new ReactShallowRenderer();
  renderer.render(<Header />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
*/
