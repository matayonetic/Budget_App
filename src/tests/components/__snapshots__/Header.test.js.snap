// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header>
  <h1>
    Budget App
  </h1>
  <NavLink
    to="/"
  >
    Home
  </NavLink>
    
  <NavLink
    to="/create"
  >
    Add Expense
  </NavLink>
    
  <NavLink
    to="/help"
  >
    Help
  </NavLink>
   
  <br />
</header>
`;
